src/tests/TyroSharedContext.spec.tsx

```diff
@@ -27,51 +27,313 @@ const renderWithProvider = async (component, options: TyroPayOptionsProps): Prom
+            `ErrorMessage: ${TyroErrorMessages[ErrorMessageType.MISSING_MERCHANT_CONFIG].message}`
+          )
+        ).not.toBeNull();
         expect(wrapper.queryByText('Or pay with card')).toBeNull();
-        const button = await wrapper.findByTestId('test-button');
-        await fireEvent.press(button);
-        expect(await wrapper.findByText('ErrorMessage: TyroProvider not initialised')).not.toBeNull();
+        expect(wrapper.queryByPlaceholderText('Card number')).toBeNull();
+        expect(wrapper.queryByPlaceholderText('Name on card')).toBeNull();
+        expect(wrapper.queryByPlaceholderText('MM/YY')).toBeNull();
+        expect(wrapper.queryByPlaceholderText('CVV')).toBeNull();
+      });
+    }, 15000);
+
+    test('TyroProvider does initialise when applePay enabled with merchantIdentifier', async () => {
+      global.fetch = jest.fn(() =>
+        mockFetch(200, { status: 'AWAITING_PAYMENT_INPUT', isLive: false } as ClientPayRequestResponse)
+      );
+      mockedHelpers.isAndroid.mockReturnValue(false);
+      mockedHelpers.isiOS.mockReturnValue(true);
+      await act(async () => {
+        await waitFor(async () => {
+          wrapper = await renderWithProvider(<InitTestComponent />, {
+            liveMode: false,
+            options: { applePay: { enabled: true, merchantIdentifier } },
+          });
+        });
+        expect(
+          await wrapper.queryByText(
+            `ErrorMessage: ${TyroErrorMessages[ErrorMessageType.MISSING_MERCHANT_CONFIG].message}`
+          )
+        ).toBeNull();
         expect(wrapper.queryByText('Or pay with card')).toBeNull();
         expect(wrapper.queryByPlaceholderText('Card number')).toBeNull();
         expect(wrapper.queryByPlaceholderText('Name on card')).toBeNull();
         expect(wrapper.queryByPlaceholderText('MM/YY')).toBeNull();
         expect(wrapper.queryByPlaceholderText('CVV')).toBeNull();
       });
-      spy.mockRestore();
     }, 15000);
-    test('Able to init and display PaySheet', async () => {
+
+    test('TyroProvider does initialise when googlePay enabled with merchantName', async () => {
+      global.fetch = jest.fn(() =>
+        mockFetch(200, { status: 'AWAITING_PAYMENT_INPUT', isLive: false } as ClientPayRequestResponse)
+      );
+      mockedHelpers.isAndroid.mockReturnValue(false);
+      mockedHelpers.isiOS.mockReturnValue(true);
+      await act(async () => {
+        await waitFor(async () => {
+          wrapper = await renderWithProvider(<InitTestComponent />, {
+            liveMode: false,
+            options: { googlePay: { enabled: true, merchantName } },
+          });
+        });
+        expect(
+          await wrapper.queryByText(
+            `ErrorMessage: ${TyroErrorMessages[ErrorMessageType.MISSING_MERCHANT_CONFIG].message}`
+          )
+        ).toBeNull();
+        expect(wrapper.queryByText('Or pay with card')).toBeNull();
+        expect(wrapper.queryByPlaceholderText('Card number')).toBeNull();
```

Comment:
should those be Not Null when TyroProvider initialises successfully? Perhaps using wrapper.getByText()